### PR TITLE
[STAT] Integrate .env file

### PR DESCRIPTION
**Pull request by TheHarvesterPM**
1) Environments were moved from src/app/environments to src/app/assets/config.[ENVIRONMENT].json so they can be loaded at runtime vs compile time. app.config.ts was added to help digest runtime environment.

2) Procfile and heroku-web.sh were added so that Heroku would copy the appropriate env file to be used based on the DEPLOYMENT_ENV set in the Heroku job Config Vars

3) src/app/main.ts was updated to pull in the environment file (assets/config.json) and create a provider called APP_CONFIG so that the environment can be injected as needed

### CODE REVIEW

File: src/app/services/prices-calculation.service.ts [hunk 1]
```diff
@@ -27,8 +27,14 @@ export class PricesCalculationService {
   private rewardEnded = new Set<string>();
   private lastPrices = new Map<string, PricesDto>();
 
-  constructor(private log: NGXLogger) {
+  constructor(private log: NGXLogger, @Inject(APP_CONFIG) public config: AppConfig, private logger: NGXLogger) {
     StaticValues.vaults.forEach(v => this.tvls.set(v, 0.0));
+    this.logger.updateConfig({
```
Comment (belbix) [inline src/app/services/prices-calculation.service.ts:32]: looks like we already did it in app.ts

File: src/app/services/websocket.service.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 import * as SockJS from 'sockjs-client';
-import {environment} from '../../environments/environment';
 import {filter, first, switchMap} from 'rxjs/operators';
 import {BehaviorSubject} from 'rxjs/internal/BehaviorSubject';
 import {WsConsumer} from './ws-consumer';
+import { AppConfig, APP_CONFIG } from 'src/app.config';
 
-export const WS_ENDPOINT = environment.wsEndpoint;
-export const RECONNECT_INTERVAL = environment.wsReconnectInterval;
+export var WS_ENDPOINT
```
Comment (belbix) [inline src/app/services/websocket.service.ts:10]: it is old variables, please remove it and use the config directly

File: src/app/services/prices-calculation.service.ts (hunk 1 above)
Comment (TheHarvesterPM) [reply src/app/services/prices-calculation.service.ts:32]: I agree that might fit better in a different place. I'm not sure where exactly, but I'm happy to move it if there are any suggestions.

Unfortunately, I was unable to get the values in app.ts to work. I think it is due to the logger being an import and the APP_CONFIG value not being set at import time. I had to add the logger.updateConfig piece here to get the correct values in the logger. Prior to adding the updateConfig the logger was outputting "undefined" values due to the APP_CONFIG not containing any values at the time of import.

I can update the values in app.ts to clearly show they are placeholder values until the APP_CONFIG value is populated and can be injected in a logger.updateConfig() call

File: src/app/services/websocket.service.ts (hunk 2 above)
Comment (TheHarvesterPM) [reply src/app/services/websocket.service.ts:10]: I will make this change

Comment (TheHarvesterPM) [reply src/app/services/websocket.service.ts]: fixed

File: src/app/services/prices-calculation.service.ts (hunk 1 above)
Comment (TheHarvesterPM) [reply src/app/services/prices-calculation.service.ts]: updated the values in app.ts to indicate they are just placeholders there and that they will be updated at runtime

Comment (belbix) [reply src/app/services/prices-calculation.service.ts]: hm, it should work in app.ts
I'll investigate this, thanks

**Pull request merged**